bigtop-packages/src/charm/hadoop/layer-hadoop-zookeeper/reactive/zookeeper.py

```diff
@@ -0,0 +1,64 @@
+'''
+Reactive handlers for Zookeeper.
+'''
+
+from charmhelpers.core import hookenv
+from charms.layer.zookeeper import Zookeeper
+from charms.reactive import set_state, when, when_not
+
+
+@when('bigtop.available')
+@when_not('zookeeper.installed')
+def install_zookeeper():
+    '''
+    After Bigtop has done the initial setup, trigger a puppet install,
+    via our Zooekeeper library.
+
+    puppet will start the service, as a side effect.
+
+    '''
+    hookenv.status_set('waiting', 'Installing Zookeeper')
```

Comment:
I am sorry about that I misled you. When we are doing things within the charm we use 'maintenance' and not 'waiting' eg hookenv.status_set('maintenance', 'installing namenode')

Sorry again...